CMakeLists.txt

```diff
@@ -15,54 +15,89 @@ if(CMAKE_CXX_COMPILER_ID MATCHES "GNU" OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   add_compile_options(-Wall -Wextra -Wpedantic)
 endif()
 
-find_package(GDAL 3 REQUIRED)
 find_package(ament_cmake REQUIRED)
+find_package(grid_map_msgs REQUIRED)
 find_package(grid_map_ros REQUIRED)
 find_package(grid_map_core REQUIRED)
-find_package(Eigen3 REQUIRED)
+find_package(rclcpp REQUIRED)
 
-add_library(${PROJECT_NAME})
-add_subdirectory(src)
+find_package(Eigen3 REQUIRED)
+find_package(GDAL 3 REQUIRED)
 
-target_include_directories(${PROJECT_NAME}
-    PUBLIC
-    "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
-    "$<INSTALL_INTERFACE:include/${PROJECT_NAME}>"
-)
+# macOS
+if(APPLE)
+  link_directories(/opt/homebrew/opt/yaml-cpp/lib)
+endif()
 
 # Reverse compatability for GDAL<3.5
 # https://gdal.org/development/cmake.html
 if(NOT TARGET GDAL::GDAL)
-    add_library(GDAL::GDAL ALIAS ${GDAL_LIBRARY})
+  add_library(GDAL::GDAL ALIAS ${GDAL_LIBRARY})
 endif()
 
-target_link_libraries(${PROJECT_NAME} Eigen3::Eigen GDAL::GDAL grid_map_core::grid_map_core)
+# Build
+add_library(${PROJECT_NAME}
+  src/grid_map_geo.cpp
+)
+
+target_include_directories(${PROJECT_NAME}
+  PUBLIC
+  "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
+  "$<INSTALL_INTERFACE:include/${PROJECT_NAME}>"
+  ${GDAL_INCLUDE_DIR}
```

Comment:
I don't think this is needed because the GDAL::GDAL target should already have all the properties. If it doesn't, let's add it up where the GDAL::GDAL alias is created.

Reply:
Removed in https://github.com/ethz-asl/grid_map_geo/commit/766e75cd831725f964b5d91519a13c77a6021070